File: f579.c
#include <stdio.h>

int main(void) {
    int a, b, n, input, ca = 0, cb = 0, ans = 0;
    scanf("%d %d", &a, &b);
    scanf("%d", &n);
    for (int i = 0; i < n; i++) {
        ca = 0;
        cb = 0;
        scanf("%d", &input);
        while (input != 0) {
            if (input == a) {
                ca++;
            }
            if (input == -a) {
                ca--;
            }
            if (input == b) {
                cb++;
            }
            if (input == -b) {
                cb--;
            }
            scanf("%d", &input);
        }
        if (ca > 0 && cb > 0) {
            ans++;
        }
    }
    printf("%d\n", ans);
    return 0; 
}

File: f579.py
# -*- coding: utf-8 -*-
a, b = map(int, input().split())
n = int(input())
ans = 0
for _ in range(n):
    t = [int(x) for x in input().split()]
    ca = t.count(a) - t.count(-a)
    cb = t.count(b) - t.count(-b)
    if (ca > 0 and cb > 0):
        ans += 1
print(ans)